### ISSUE TITLE
Final snapshot no longer available after TeamCity 2019.1

### ISSUE THREAD

**Issue by DanWallaceAdslot** [JetBrains/TeamCity.SonarQubePlugin#62]
TL;DR:
Build configuration using Nunit3 to run tests with dotCover as coverage tool
* TeamCity 2018.x -> final .dcvr snapshot was generated and able to be stored as an artifact
* TeamCity 2019.1 -> final .dcvr is no longer available and can't be stored as an artifact

I was using a basic setup:
* Build configuration that ran my build (VS2017 runner) then a second step with NUnit3 and having dotCover as the coverage tool. Saved the resulting .dcvr as an artifact
* Second build configuration with snapshot+artifact dependency, used that .dcvr to generate a html report, then used that in the parameters for dotCover

This was allowing me to see the dotCover results in TeamCity and also the only way I could figure out how to send coverage through to SonarQube.

Was working fantastically until I upgraded TeamCity to 2019.1. Now the .dcvr file is missing and am currently in the process of trying to figure out an alternative method

**Comment by Linfar** [comment]
Yes, since 2019.1 .dcvr file is not published by default. Please add a `teamcity.agent.dotCover.publishSnapshot=true` config parameter. You can also use `teamcity.agent.dotNetCoverage.publishPath` parameter to specify file location.

**Comment by DanWallaceAdslot** [comment]
Thanks Linfar, I've done that and the file is being generated again :)
However, the runner is unable to parse the resuling html file

I assume this is because of linebreaks? Looking at the file, it's fine

```C:\TeamCity\buildAgent\tools\JetBrains.dotCover.CommandLineTools.2019.1.1\dotCover.exe report /ReportType=HTML /Source="C:\work\125ab98fa6e9b045/dotCover.dcvr" /Output="C:\work\125ab98fa6e9b045/dotCover.html"```

![image](https://user-images.githubusercontent.com/25143022/59814599-31e56380-9359-11e9-8353-6306418f7d94.png)

![image](https://user-images.githubusercontent.com/25143022/59814660-707b1e00-9359-11e9-91e7-94979e7e64ec.png)

Should I move this to a new issue and close this one off?

**Comment by Linfar** [comment]
Thanks for update

**Comment by nazarenomanco** [comment]
@Linfar please help me, where do I have to add "teamcity.agent.dotCover.publishSnapshot=true" config parameter? 
May you please make me an example. I do not find the correct point. I'm using the  2019.2 version 
Thank you

**Comment by Linfar** [comment]
@nazarenomanco 
You should add a [build parameter](https://www.jetbrains.com/help/teamcity/configuring-build-parameters.html#ConfiguringBuildParameters-DefiningBuildParametersinBuildConfiguration) for the Build Configuration. Navigate to Edit Build Configuration administration page, find Parameters tab and push Add new parameter button.

**Comment by nazarenomanco** [comment]
@Linfar thank you! I have added here the parameters but it give me an error. There is any tutorial somewhere?
Thanks!

![image](https://user-images.githubusercontent.com/11135338/72049328-00b18e00-32bf-11ea-8cf8-44c10aa09c4b.png)

**Comment by Linfar** [comment]
What error? I don't see neither parameter nor error on your screenshot
